<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
  <meta charset="utf-8">
  <title>RNA-seq基本概念和数据分析流程 - 牛牛龙</title>
  <meta name="author" content="Yulong Niu">

  
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis">
  <link href="/favicon.ico" type="image/png" rel="icon">
  <link href="/atom.xml" rel="alternate" title="牛牛龙" type="application/atom+xml">

  <!-- http://opengraphprotocol.org/ -->
  <meta name="twitter:card" content="summary_large_image">
  <meta property="og:type" content="website">
  <meta property="og:url" content="http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis">
  <meta property="og:title" content="RNA-seq基本概念和数据分析流程 - 牛牛龙">
  

  <!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({ TeX: { extensions: ["color.js"] }});
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

<script src="/javascripts/libs/jquery/jquery-2.0.3.min.js"></script>

<!--enable to choose theme in octostrap-->

  <link href="/assets/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/assets/bootstrap/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">



<!-- link open with new tab  -->
<script>
  function addBlankTargetForLinks () {
  $('a[href^="http"]').each(function(){
  $(this).attr('target', '_blank');
  });
  }
  
  $(document).bind('DOMNodeInserted', function(event) {
  addBlankTargetForLinks();
  });
</script>


  
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  
  <link href="/stylesheets/data-table.css" media="screen, projection" rel="stylesheet" type="text/css" />
  

  
   <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-53073828-1', 'auto');
    ga('send', 'pageview');

  </script>


</head>

  <body   >
    <a href="#content" class="sr-only sr-only-focusable">Skip to main content</a>
    <div id="wrap">
      <header role="banner">
        <nav class="navbar navbar-default" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">牛牛龙</a>
        </div>

        <div class="navbar-collapse collapse">
            <ul class="nav navbar-nav">
                <li class="active">
                    <a rel="index" href="/">Blog</a>
                </li>
                <li >
                    <a href="/blog/archives">Archives</a>
                </li>
		<li >
                  <a href="/whoami">WhoAmI</a>
                </li>
		<li >
                  <a href="/link">Link</a>
                </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="subscribe-rss" href="/atom.xml" title="subscribe via RSS">
                        <span class="visible-xs">RSS</span>
                        <img class="hidden-xs" src="/images/rss.png" alt="RSS">
                    </a>
                </li>
                
            </ul>
            
                <form class="search navbar-form navbar-right" action="https://www.google.com/search" method="GET">
                    <input type="hidden" name="q" value="site:yulongniu.bionutshell.org">
                    <div class="form-group">
                        <input class="form-control" type="text" name="q" placeholder="Search">
                    </div>
                </form>
            
        </div>
    </div>
</nav>


      </header>
      <div id="main" role="main" class="container">
        <div id="content">
          <div class="row">
  <div class="page-content col-md-9" itemscope itemtype="http://schema.org/Blog">
    <meta itemprop="name" content="牛牛龙" />
    
    <meta itemprop="url" content="http://yulongniu.bionutshell.org" />
    <article class="hentry" role="article" itemprop="blogPost" itemscope itemtype="http://schema.org/BlogPosting">
      
  <header class="page-header">
    
      <p class="meta text-muted text-uppercase">
        












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-23T05:03:15+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
        
           | <a href="#disqus_thread" itemprop="discussionUrl"
             data-disqus-identifier="http://yulongniu.bionutshell.org">Comments</a>
        
      </p>
    
    
    <h1 class="entry-title" itemprop="name headline">
        RNA-seq基本概念和数据分析流程
        
    </h1>
    
  </header>


<div class="entry-content clearfix" itemprop="articleBody description"><p>RNA-seq相比较基因芯片，价格虽然昂贵一些，精度和灵敏度更高。同时，在测序深度足够时，也可以检测mRNA选择性剪切类型。</p>

<h2 id="section">1. 样品制备</h2>

<!--more-->

<h2 id="section-1">2. 测序</h2>

<p>Illumina双端测序<a href="http://www.illumina.com/technology/next-generation-sequencing/paired-end-sequencing_assay.html">动画</a>和<a href="http://onetipperday.blogspot.sg/2013/12/illumina-hiseq2000-adapter-and.html">图示</a>。</p>

<h2 id="section-2">3. 分析流程</h2>

<p>RNA-seq<a href="http://blog.qiuworld.com:8080/archives/2343">主要分析流程</a>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Workflow of RNA-seq data analysis </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line">原始数据质量评估 --&gt;
</span><span class="line">数据清洗（去除接头和低质量read） --&gt;
</span><span class="line">清洗数据质量评估 --&gt;
</span><span class="line">map测序结果至基因组（转录组） --&gt;
</span><span class="line">map数据质量评估 --&gt;
</span><span class="line">差异表达基因/选择性剪切/新基因/融合基因选择 --&gt;
</span><span class="line">GO和pathway分析 --&gt;
</span><span class="line">共表达网络分析
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h3 id="section-3">3.1 序列清洗</h3>

<p>序列清洗主要是去除测序结果中的<a href="http://onetipperday.blogspot.sg/2013/06/illumina-hiseq2000-adapter.html">adapter</a>或通用引物等。</p>

<p>以下使用Illumina HiSeq2000平台，对一个人类样本的RNA测序。统计各种序列清洗方法和选择后的reads数目。原始数据两端测序reads分别为54,492,228和54,492,228。</p>

<hr />

<table>
  <thead>
    <tr>
      <th>Method</th>
      <th>#Trimmed</th>
      <th>#Mapped*</th>
      <th>#Filtered</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>r50-notrim</td>
      <td>108,984,456</td>
      <td>109,278,388</td>
      <td>79,143,942</td>
    </tr>
    <tr>
      <td>r50-nomixed-notrim</td>
      <td>108,984,456</td>
      <td>103,548,800</td>
      <td>79,143,942</td>
    </tr>
    <tr>
      <td>r50-nomixed-trimmomatic-min20</td>
      <td>104,164,622</td>
      <td>116,315,394</td>
      <td>80,337,256</td>
    </tr>
    <tr>
      <td>r50-nomixed-trimmomatic-min36</td>
      <td>101,548,172</td>
      <td>110,778,108</td>
      <td>79,248,896</td>
    </tr>
    <tr>
      <td>r50-nomixed-trimmomatic-min50</td>
      <td>98,525,312</td>
      <td>106,659,988</td>
      <td>77,779,424</td>
    </tr>
    <tr>
      <td>r50-nomixed-galore-min20</td>
      <td>107,097,862</td>
      <td>114,943,386</td>
      <td>83,039,928</td>
    </tr>
    <tr>
      <td>r100-nomixed-galore-min20</td>
      <td>107,097,862</td>
      <td>114,944,672</td>
      <td>87,899,316</td>
    </tr>
    <tr>
      <td>r165sd45-nomixed-galore-min20</td>
      <td>107,097,862</td>
      <td>114,201,366</td>
      <td>90,750,208</td>
    </tr>
    <tr>
      <td>r165sd45G-nomixed-galore-min20</td>
      <td>107,097,862</td>
      <td>109,901,742</td>
      <td>93,477,122</td>
    </tr>
    <tr>
      <td>r165sd45-nomixed-galore-min50</td>
      <td>104,869,208</td>
      <td>109,258,544</td>
      <td>89,329,672</td>
    </tr>
  </tbody>
</table>

<p><strong>*</strong>：使用TopHat2把序列mapped到hs19基因组。TopHat2默认设置为，如果一个reads能mapped到多个位点，则都会报道。因此数目可能比原始数据多。</p>

<hr />

<h2 id="section-4">4. 分析过程注意事项</h2>

<ol>
  <li>参考转录组注释文件（GFF/GTF）的染色体编号，与基因组信息一致。比如都用<code class="language-bash">chr1</code>，或者都用<code class="language-bash">1</code>标识1号染色体。可能出现问题地方：</li>
</ol>

<blockquote>
  <ul>
    <li>
      <p><a href="http://ccb.jhu.edu/software/tophat/manual.shtml">TopHat</a>的<code class="language-bash">-G/--GTF</code>参数。</p>
    </li>
    <li>
      <p><a href="http://miso.readthedocs.org/en/fastmiso/#human-mouse-gene-models-for-isoform-centric-analyses">MISO</a>程序<code class="language-bash">pe_utils</code>的传入转录组注释文件。</p>
    </li>
  </ul>

</blockquote>

<h2 id="section-5">5. 基本概念</h2>

<ul>
  <li>
    <p><strong>Read</strong>：是组成测序的结果的基本单位。</p>
  </li>
  <li>
    <p><strong>Count</strong>：在某次测序中，对于某个指标（比如某个基因），得到的reads数的总和。</p>
  </li>
  <li>
    <p><strong>RPKM</strong>（Reads Per Kilobase per Million）和<strong>FPKM</strong>（Fragments Per Kilobase per Million）：</p>
  </li>
</ul>

<blockquote>
  <p>首先需要解释FPKM和RPKM的原理是相似的，区别在于FPKM对应的是DNA片段，比如在一个Illumina的pair-end（双尾）RNA-seq中，一对（两个）reads对应是一个DNA片段。有了FPKM（RPKM）概念，我们就能比较：同一个样本中基因A和基因B的相对表达量；或者不同样本中，同一个基因的相对表达量。</p>

  <p>具体的原因是：引入“每一千碱基（per kilobase）”的原因在于，不同的RNA可能有不同长度，长度越长，对应的reads就越多。当每个RNA都除以自身长度（以1000碱基为单位）时，就可以比较同一个样本中不同基因的相对表达量了。相似地，引入“每一百万reads”的原因是，不同的样本可能测序的深度不一样，深度越深，当然对应的reads就越多了。如果结果除以各自库的数量（以一百万reads为单位），那么我们就能很好地衡量两个不同样本中同一个基因的相对表达量。</p>
</blockquote>

<ul>
  <li>
    <p><strong>Alignment</strong>：确定测得的reads在基因组上的位置的过程。</p>
  </li>
  <li>
    <p><strong>Mapping</strong>：确定aligned reads对应的转录本。</p>
  </li>
  <li>
    <p><strong>Pair end (PE)</strong>和<strong>Mate-Pair (MP)</strong>：</p>
  </li>
</ul>

<blockquote>
  <p>两种双端测序的方法，主要区别在样品库制备和测序上。比如PE制备的库是adapter在目标序列两端，而MP库中adapter在目标序列中间。因此，在数据分析时，MP类型测序必须注意剔除adapter。具体参考<a href="http://seqanswers.com/forums/showthread.php?t=503">论坛讨论</a>和<a href="http://scottmyourstone.blogspot.sg/2013/11/difference-between-paired-end-and-mate.html">Difference Between Paired-End and Mate-Pair Reads</a>。</p>

  <p>在序列软件软件中，有时也称呼一对测序reads（对同一个目标片段分别测得的正义链和反义链），它们互为mate，分别存在两个对应的测序结果文件中。</p>
</blockquote>

<ul>
  <li><strong>Adapter（接头）</strong>、<strong>Barcode（标签）</strong>和<strong>Insert（插入片段）</strong>：</li>
</ul>

<blockquote>
  <p>adapter是一段短的序列已知的核酸链，用于链接序列未知的目标测序片段。</p>

  <p>barcode，也称为index，是一段很短的寡居核酸链，用于在多个样品混合测序时，标记不同的样品。
insert是用于测序的目标片段，因为是包括在两个adapter之间，所以被称为“插入”片段。</p>

  <p>一个常见测序片段类似与<code class="language-bash">adapter--barcode--insert--adapter</code>。测序开始时前几个碱基无法测得，第一个adapter在数据输出时被去除；由于测序仪读长限制，第二个adapter通常无法测得。所以，经常得到类似 <code class="language-bash">barcode--部分insert</code>的read。最后，把barcode去除，只保留测度insert的片段，这个操作的术语是demultiplexing。</p>

  <p>需要注意的是，<a href="http://www.illumina.com/documents/products/datasheets/datasheet_truseq_sample_prep_kits.pdf">Illumina TruSeq</a>样品库制备方法中，barcode是在adapter中部，而且是与insert分开测序。而<a href="https://support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/samplepreps_nextera/nexteramatepair/nextera-mate-pair-reference-guide-15035209-02.pdf">Illumina Nextera Mate Pair</a>样品库制备中，adapter在目标序列中部。</p>
</blockquote>

<ul>
  <li>
    <p><strong>Concordant Pairs</strong>和<strong>Discordant Pairs</strong>：根据<a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#paired-inputs">Bowtie2</a>的解释，concordant pairs表示一对reads在alignment时，既方向匹配又有合适距离（Bowtie2中是200bp～500bp）。如果上述方向和距离，任意一个条件不满足，则称为discordant pairs。</p>
  </li>
  <li>
    <p><strong>chrN_random</strong>和<strong>chrUn</strong>：基因组文件中通常含有类似<code class="language-bash">chr9_gl000198_random</code>和<code class="language-bash">chrUn_gl000211</code>的基因组。根据<a href="https://genome.ucsc.edu/FAQ/FAQdownloads.html#download11">UCSC</a>解释，<code class="language-bash">chrN_random</code>包括基因组已知但具体位置未知序列，或者位置已知但具体内容未完成序列。<code class="language-bash">ChrUn</code>中包括一些具体位置未知的序列。</p>
  </li>
  <li>
    <p><strong>chrN_xxx_hap1</strong>：转录组注释文件中会出现类似<code class="language-bash">chr6_apd_hap1</code>和<code class="language-bash">chr6_dbb_hap3</code>的基因组注释。根据<a href="http://genome.ucsc.edu/cgi-bin/hgGateway?org=Human&amp;db=hg19">UCSC解释</a>这些基因组是<a href="http://hapmap.ncbi.nlm.nih.gov/originhaplotype.html.en">单倍型(haplotype)基因组</a>。</p>
  </li>
</ul>

<h3 id="section-6">参考网址</h3>

<ul>
  <li>
    <p>推荐幻灯片<a href="http://www.slideshare.net/mikaelhuss/all-bio-rnaseqqc?from_action=save">RNA-seq quality control and pre-processing</a></p>
  </li>
  <li>
    <p><a href="https://www.bioconductor.org/help/workflows/rnaseqGene/">Bioconductor详细流程</a></p>
  </li>
  <li>
    <p>RPKM和FPKM：<a href="http://jefworks.com/rpkm-and-fpkm-explained/">1</a>、 <a href="http://www.cureffi.org/2013/09/12/counts-vs-fpkms-in-rna-seq/">2</a>和<a href="http://www.partek.com/Tutorials/microarray/User_Guides/UnderstandingReads.pdf">3一个计算RPKM的例子</a></p>
  </li>
  <li>
    <p>ENCODE推荐的RNA-seq数据分析指导 <a href="http://genome.ucsc.edu/ENCODE/protocols/dataStandards/ENCODE_RNAseq_Standards_V1.0.pdf">The ENCODE Consortium: Standards, Guidelines and Best Practices for RNA-seq</a></p>
  </li>
  <li>
    <p>RNA-seq差异表达分析工作流程：<a href="http://216.49.144.90:8080/archives/3007/comment-page-2">中文</a>，<a href="http://vallandingham.me/RNA_seq_differential_expression.html">英文</a></p>
  </li>
  <li>
    <p><a href="http://www.rna-seqblog.com/how-many-reads-are-enough/">多少个read才够</a></p>
  </li>
  <li>
    <p><a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#paired-inputs">Bowtie2对一些常用名词解释</a></p>
  </li>
  <li>
    <p><a href="http://www.plob.org/2014/11/09/8672.html">二代测序中barcodes index的介绍</a></p>
  </li>
  <li>
    <p>Illumina样品制备参考：<a href="https://www.med.unc.edu/pharm/calabreselab/files/tufts-sequencing-primer">Illumina TruSeq DNA Adapters De-Mystified</a>和<a href="http://bioinformatics.cvr.ac.uk/blog/illumina-adapter-and-primer-sequences/">Illumina adapter and primer sequences</a></p>
  </li>
  <li>
    <p><a href="http://cgrlucb.wikispaces.com/file/view/readMapping.pdf">Best Practices and Suggestions for Read Alignment</a></p>
  </li>
  <li>
    <p><a href="http://barcwiki.wi.mit.edu/wiki/SOPs/rna-seq-diff-expressions">Using RNA-seq to quantify gene levels and assay for differential expression</a></p>
  </li>
</ul>

<h3 id="section-7">更新记录</h3>

<p>2017年9月16日</p>
</div>


      <footer>
        <p class="meta text-muted">
          
  

<span class="glyphicon glyphicon-user"></span> <span class="byline author vcard" itemprop="author" itemscope itemtype="http://schema.org/Person">Posted by <span class="fn" itemprop="name">Yulong Niu</span></span>

          












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-23T05:03:15+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
          

<span class="glyphicon glyphicon-tags"></span>&nbsp;
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinfor/'>bioinfor</a>
  
</span>


        </p>
        
          <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis/" data-via="" data-counturl="http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis/" >Tweet</a>
  
  
  
</div>

        
        
          <ul class="meta text-muted pager">
            
            <li class="previous"><a href="/blog/2014/07/23/octopress-install/" title="Previous Post: Octopress安装和使用">&laquo; Octopress安装和使用</a></li>
            
            
            <li class="next"><a href="/blog/2014/07/26/short-sequence-alignment/" title="Next Post: 二代测序中的短序列比对">二代测序中的短序列比对 &raquo;</a></li>
            
          </ul>
        
      </footer>
    </article>
    
      <section>
        <h2>Comments</h2>
        <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
      </section>
    
  </div>

  
  <aside class="sidebar col-md-3">
    
      <section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Recent Posts</h3>
  </div>
  
  <div id="recent_posts" class="list-group">
    
    <a class="list-group-item " href="/blog/2017/10/24/bray-curtis-distance/">Bray-Curtis Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/22/google-distance/">Normalized Google Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/16/max-entropy/">最大熵模型</a>
    
    <a class="list-group-item " href="/blog/2017/10/15/naive-bayes-binary-multinomial/">朴素贝叶斯分类器应用于二元数据类型</a>
    
    <a class="list-group-item " href="/blog/2017/06/14/shadowsocks-setup/">Shadowsocks设置推荐</a>
    
  </div>
</section>
<section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Categories</h3>
  </div>
  <div class="list-group">
    
    
    <a class="list-group-item " href="/blog/categories/linux/index.html">
        <span class="badge">9</span>
        linux
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/r/index.html">
        <span class="badge">9</span>
        r
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/lisp/index.html">
        <span class="badge">6</span>
        lisp
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/ppr/index.html">
        <span class="badge">5</span>
        ppr
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/bioinfor/index.html">
        <span class="badge">19</span>
        bioinfor
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/saysomething/index.html">
        <span class="badge">1</span>
        saysomething
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/c/index.html">
        <span class="badge">2</span>
        c
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/hcj/index.html">
        <span class="badge">1</span>
        hcj
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/share/index.html">
        <span class="badge">1</span>
        share
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/java/index.html">
        <span class="badge">1</span>
        java
      </a>
    
  </div>
</section>






    
  </aside>
  
</div>

        </div>
      </div>
    </div>
    <footer role="contentinfo"><div class="container">
    <p class="text-muted credits">
  Copyright &copy; 2018 - Yulong Niu<br>
  <small>
      <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>,
      <span class="credit">customized with <a href="https://github.com/kAworu/octostrap3">octostrap3</a></span>.
  </small>
</p>

</div>
</footer>
    

<script type="text/javascript">
      var disqus_shortname = 'yulongniu';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis/';
        var disqus_url = 'http://yulongniu.bionutshell.org/blog/2014/07/23/rna-seq-analysis/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>


<script src="/assets/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="/javascripts/modernizr-2.0.js"></script>


  </body>
</html>
