<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
  <meta charset="utf-8">
  <title>TopHat/Cufflinks/CummeRbund使用介绍 - 牛牛龙</title>
  <meta name="author" content="Yulong Niu">

  
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund">
  <link href="/favicon.ico" type="image/png" rel="icon">
  <link href="/atom.xml" rel="alternate" title="牛牛龙" type="application/atom+xml">

  <!-- http://opengraphprotocol.org/ -->
  <meta name="twitter:card" content="summary_large_image">
  <meta property="og:type" content="website">
  <meta property="og:url" content="http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund">
  <meta property="og:title" content="TopHat/Cufflinks/CummeRbund使用介绍 - 牛牛龙">
  

  <!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({ TeX: { extensions: ["color.js"] }});
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
 MathJax.Hub.Queue(function() {
     var all = MathJax.Hub.getAllJax(), i;
     for(i=0; i < all.length; i += 1) {
         all[i].SourceElement().parentNode.className += ' has-jax';
     }
 });
</script>

<script type="text/javascript"
        src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

<script>
 MathJax.Hub.Config({
     config: ["MMLorHTML.js"],
     extensions: ["tex2jax.js","TeX/AMSmath.js","TeX/AMSsymbols.js"],
     jax: ["input/TeX"],
     tex2jax: {
         inlineMath: [ ['$','$'], ["\\(","\\)"] ],
         displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
         processEscapes: false
     },
     TeX: {
         TagSide: "right",
         TagIndent: ".8em",
         MultLineWidth: "85%",
         equationNumbers: {
             autoNumber: "AMS",
         },
         unicode: {
             fonts: "STIXGeneral,'Arial Unicode MS'"
         }
     },
     showProcessingMessages: false
 });
</script>

<script src="/javascripts/libs/jquery/jquery-2.0.3.min.js"></script>

<!--enable to choose theme in octostrap-->

  <link href="/assets/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/assets/bootstrap/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">



<!-- link open with new tab  -->
<script>
  function addBlankTargetForLinks () {
  $('a[href^="http"]').each(function(){
  $(this).attr('target', '_blank');
  });
  }

  $(document).bind('DOMNodeInserted', function(event) {
  addBlankTargetForLinks();
  });
</script>


  
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  

  
   <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-53073828-1', 'auto');
    ga('send', 'pageview');

  </script>


</head>

  <body   >
    <a href="#content" class="sr-only sr-only-focusable">Skip to main content</a>
    <div id="wrap">
      <header role="banner">
        <nav class="navbar navbar-default" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">牛牛龙</a>
        </div>

        <div class="navbar-collapse collapse">
            <ul class="nav navbar-nav">
                <li class="active">
                    <a rel="index" href="/">Blog</a>
                </li>
                <li >
                    <a href="/blog/archives">Archives</a>
                </li>
		<li >
                  <a href="/whoami">WhoAmI</a>
                </li>
		<li >
                  <a href="/link">Link</a>
                </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="subscribe-rss" href="/atom.xml" title="subscribe via RSS">
                        <span class="visible-xs">RSS</span>
                        <img class="hidden-xs" src="/images/rss.png" alt="RSS">
                    </a>
                </li>
                
            </ul>
            
                <form class="search navbar-form navbar-right" action="https://www.google.com/search" method="GET">
                    <input type="hidden" name="q" value="site:yulongniu.bionutshell.org">
                    <div class="form-group">
                        <input class="form-control" type="text" name="q" placeholder="Search">
                    </div>
                </form>
            
        </div>
    </div>
</nav>


      </header>
      <div id="main" role="main" class="container">
        <div id="content">
          <div class="row">
  <div class="page-content col-md-9" itemscope itemtype="http://schema.org/Blog">
    <meta itemprop="name" content="牛牛龙" />
    
    <meta itemprop="url" content="http://yulongniu.bionutshell.org" />
    <article class="hentry" role="article" itemprop="blogPost" itemscope itemtype="http://schema.org/BlogPosting">
      
  <header class="page-header">
    
      <p class="meta text-muted text-uppercase">
        












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-06-26T14:51:56+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
        
           | <a href="#disqus_thread" itemprop="discussionUrl"
             data-disqus-identifier="http://yulongniu.bionutshell.org">Comments</a>
        
      </p>
    
    
    <h1 class="entry-title" itemprop="name headline">
        TopHat/Cufflinks/CummeRbund使用介绍
        
    </h1>
    
  </header>


<div class="entry-content clearfix" itemprop="articleBody description"><p><a href="http://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a>、<a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a>和cummeRbund，被称为处理RNA-seq数据的“燕尾服（tuxedo）”。TopHat负责RNA-seq的reads映射比对到基因组，并且自动识别mRNA“内含子-外显子”剪切；Cufflinks擅长组装转录组和寻找差异表达基因（或转录起始位点TSS等）；cummeRbund主攻数据可视化。</p>

<!--more-->

<h2 id="tophat">1. Tophat</h2>

<p><strong>简介</strong>：<a href="http://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a>是快速将RNA测序片段“对应（mapping）”到基因组上的工具，优势在于处理外显子间的剪切。内部首先使用bowtie或bowtie2把RNA测序片段“比对（alignment）”到基因组，之后再分析和鉴定剪切连接区域。</p>

<p><strong>平台</strong>：Mac OS/Linux</p>

<p><strong>快速运行</strong>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 双端测序</span>
</span><span class="line"><span class="nv">$ </span>tophat2 -p <span class="m">8</span> -o human_1 <span class="se">\</span>
</span><span class="line">          --mate-inner-dist <span class="m">165</span> --mate-std-dev <span class="m">45</span> --no-mixed <span class="se">\</span>
</span><span class="line">          hg19 human_1.fastq.gz human_2.fastq.gz
</span><span class="line">
</span><span class="line"><span class="c"># 单端测序</span>
</span><span class="line"><span class="nv">$ </span>tophat2 -p <span class="m">8</span> -o human_1 hg19 human_1.fastq.gz
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash">-p</code>：设置线程数，用于多核计算。</p>
  </li>
  <li>
    <p><code class="language-bash">-o</code>：执行输入文件夹。</p>
  </li>
  <li>
    <p><code class="language-bash">-r/--mate-inner-dist</code>：一对reads的间隔长度的期望平均值，默认值为50bp，<strong>建议添加</strong>。具体解释，参考<a href="http://blog.qiuworld.com:8080/archives/3007">RNA-seq差异表达分析工作流程</a>和<a href="http://www.plob.org/2012/12/04/4988.html">Tophat中-r/–mate-inner-dist参数</a>。</p>
  </li>
  <li>
    <p><code class="language-bash">--mate-std-dev</code>：一对reads的间隔长度分布的标准差，默认值为20bp，<strong>建议添加</strong>。<code class="language-bash">-r/--mate-inner-dist</code>和<code class="language-bash">--mate-std-dev</code>的估计方法参考<a href="http://yulongniu.bionutshell.org/blog/2015/05/23/tophat-inner-distance/">确定TopHat中mate-inner-dist参数</a>。</p>
  </li>
  <li>
    <p><code class="language-bash">--library-type</code>：</p>
  </li>
</ul>

<blockquote>
  <p>测序仪器和方法，默认为标准Illumina平台的<code class="language-bash">fr-unstranded</code>。其他平台设置，详见<a href="http://ccb.jhu.edu/software/tophat/manual.shtml">TopHat说明文档</a>、<a href="http://onetipperday.blogspot.sg/2012/07/how-to-tell-which-library-type-to-use.html">How to tell which library type to use (fr-firststrand or fr-secondstrand)?</a>和<a href="http://www.plob.org/2013/12/03/6731.html">链特异性转录组原理</a>。</p>

  <p>如果分不清楚<code class="language-bash">fr-firststrand</code>和<code class="language-bash">fr-secondstrand</code>，推荐两种方法：<a href="http://ccb.jhu.edu/software/tophat/faq.shtml">第一种</a>用两个参数试运行一个有1M reads的小样本，之后比较<code class="language-bash">junction.bed</code>大小；<a href="http://onetipperday.blogspot.sg/2012/07/how-to-tell-which-library-type-to-use.html">第二种</a>在两个双端测序文件（<code class="language-bash">fastq.gz</code>）中抽取一些reads，之后<a href="http://genome.ucsc.edu/cgi-bin/hgBlat?org=human">Blat</a>到USCS genomes上观察。</p>
</blockquote>

<ul>
  <li>
    <p><code class="language-bash">--no-discordant</code>：只对于paired reads，只报告concordant mappings。加入这个参数，tophat2在最后一步失败。也可以不加入这个参数，通过sam/bam文件第二列过滤discordant reads，方法参考<a href="http://yulongniu.bionutshell.org/blog/2015/05/16/filter-tophat2-output/">过滤TopHat分析双端测序的输出</a>。</p>
  </li>
  <li>
    <p><code class="language-bash">--no-mixed</code>：只对于paired reads，只报告paired reads都成功map。TopHat默认不加这个参数，即如果对于一个read，如果没有找到alignment的concordant或者discordant mate，那么这一对read将分别寻找和报道各自的alignment。这个参数与<code class="language-bash">--no-discordant</code>不同，因为加上<code class="language-bash">--no-mixed</code>也可能报道discordant pairs（例如一对reads都成功alignment，但是方向或者之间距离不对）。</p>
  </li>
  <li>
    <p><code class="language-bash$">-g/--max-multihits</code>：对于多个map的reads，设定报告数目，默认数值为20。需要注意，尽管这个参数可以设定为1，也不能用于设定唯一map的read。因为某个read有可能map到基因组多个位点，当设定为1时，只会返回得分最高的情况。</p>
  </li>
</ul>

<p><strong>后续操作</strong>：</p>

<ul>
  <li>
    <p>TopHat2运行后查看<code class="language-bash">align_summary.txt</code>获得比对结果。</p>
  </li>
  <li>
    <p>TopHat2会输出<code class="language-bash">accepted_hits.bam</code>（接受map的reads文件）和<code class="language-bash">unmapped.bam</code>（没有map上的reads文件）。对于后者，使用基因组浏览器，如<a href="http://www.broadinstitute.org/igv/">IGV</a>或者<a href="http://genome.ucsc.edu/">UCSC Genome Browser</a>大致看下是有无map，之后可以直接丢弃。</p>
  </li>
  <li>
    <p>过滤双端测序的TopHat结果，参考<a href="http://yulongniu.bionutshell.org/blog/2015/05/16/filter-tophat2-output/">过滤TopHat分析双端测序的输出</a>。</p>
  </li>
</ul>

<p><strong>补充</strong>：</p>

<ul>
  <li>
    <p>TopHat2运行时，可以将基因组的fasta格式文件（比如<code class="language-bash">hg19.fa</code>），一起放在index的文件夹中（如果没有，Bowtie2先生成一个），可以节省运行时间。具体生成方法，参考：<a href="http://yulongniu.bionutshell.org/blog/2014/07/26/short-sequence-alignment/">二代测序中的短序列比对</a>。</p>
  </li>
  <li>
    <p><a href="http://www.acgt.me/blog/2015/4/27/the-dangers-of-default-parameters-in-bioinformatics-lessons-from-bowtie-and-tophat?utm_content=bufferb2c35&amp;utm_medium=social&amp;utm_source=twitter.com&amp;utm_campaign=buffer">不加思考地使用默认参数的下场</a></p>
  </li>
</ul>

<h2 id="cufflinks">2. Cufflinks</h2>

<p><strong>简介</strong>：<a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a>是TopHat的下游工具，用于分析差异表达基因、差异转录起始位点、新基因和选择性剪切。一般可以分为三步：1. <code class="language-bash">cufflinks</code>对每个bam文件生成转录组；2. <code class="language-bash">cuffmerge</code>结合真实转录组和bam生成转录组，构建一个整合转录组；3. <code class="language-bash">cuffdiff</code>比较不同生物学样本，寻找差异表达基因。</p>

<p><strong>平台</strong>：Mac OS/Linux</p>

<h3 id="cufflinkslanguage-bash-">2.1 <code class="language-bash">cufflinks</code> 快速运行</h3>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code for cufflinks </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="nv">$ </span>cufflinks -p <span class="m">8</span> -g hg19_ensembl.gtf -b hg19.fa -M hg19_rRNAtRNAchrM -u <span class="se">\</span>
</span><span class="line">            -o outPutLinks accepted_filtered.bam
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash$">-p/--num-threads</code>：设置线程数，用于多核计算。</p>
  </li>
  <li>
    <p><code class="language-bash$">-o/--output-dir</code>：输出文件夹。</p>
  </li>
  <li>
    <p><code class="language-bash$">-g/--GTF-guide</code>：使用参考转录组注释文件指导组装过程，<strong>建议添加</strong>。加入这个参数和对应的注释文件，可以有效提高cufflinks转录本组装。输出的转录组是传入注释文件和cufflinks自行组装的联合。具体解释如下图<a href="#Roberts et.al., 2011">[Roberts et.al., 2011]</a>：</p>
  </li>
</ul>

<p><img src="/images/faux_reads_cufflinks.png" title="image" alt="cufflinks原理图" /></p>

<ul>
  <li>
    <p><code class="language-bash$">-G/--GTF</code>：只对传入的转录组注释文件进行表达量估计，即生成的结果如<code class="language-bash">genes.fpkm_tracking</code>只包括转录组注释文件，而不包括cufflinks自己可能的组装。如果RNA-seq分析的目的是为了获得差异表达基因，而不是寻找新的转录本，建议添加，以加快运行速度。</p>
  </li>
  <li>
    <p><code class="language-bash$">-b/--frag-bias-correct</code>：提供一个multifasta文件，可以提高转录丰度的准确度，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-M/–mask-file</code>：需要去除的转录本，比如tRNA、rRNA和线粒体转录本，<strong>建议添加</strong>。在RNA-seq数据中，这些转录本量非常高。<a href="http://cole-trapnell-lab.github.io/cufflinks/cufflinks/index.html">cufflinks</a>提示剔除这些转录本，有助于提高转录量估计的准确程度。tRNA、rRNA和线粒体基因组注释信息提取，参考<a href="http://yulongniu.bionutshell.org/blog/2015/06/02/download-ucsc-gene-mask/">UCSC Table下载注释文件</a>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-u/--multi-read-correct</code>：处理map到基因组多个位置的reads，<strong>建议添加</strong>。对于TopHat，建议过滤掉这部分reads，过滤方法见<a href="http://yulongniu.bionutshell.org/blog/2015/05/16/filter-tophat2-output/">过滤TopHat分析双端测序的输出</a>。按照上述方法过滤这部分reads后，则不需要加这个参数。</p>
  </li>
  <li>
    <p><code class="language-bash$">--library-type</code>：设定测序方法，默认为<code class="language-bash">fr-unstranded</code>，可以用于典型的Illumina双端（paired-end）测序。</p>
  </li>
</ul>

<h3 id="cuffmergelanguage-bash-">2.2 <code class="language-bash">cuffmerge</code> 快速运行</h3>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code for cuffmerge </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="nv">$ </span>cuffmerge -p <span class="m">8</span> -g hg19USCS_ensembl.gtf -s hg19.fa -o mergeFile assemblies.txt
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash$">-p/--num-threads </code>：设置线程数，用于多核计算。</p>
  </li>
  <li>
    <p><code class="language-bash$">-o/--output-dir</code>：输出文件夹。</p>
  </li>
  <li>
    <p><code class="language-bash$">g/--ref-gtf</code>：参考转录组，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">g/--ref-gtf</code>：参考基因组序列，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">assemblies.txt</code>：是一个文件，写入需要组装转录本的路径。</p>
  </li>
</ul>

<h3 id="cuffquantlanguage-bash-">2.3 <code class="language-bash">cuffquant</code> 快速运行</h3>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code for cuffd </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="nv">$ </span>cuffquant -p <span class="m">14</span> -o quantOut -b hg19.fa <span class="se">\</span>
</span><span class="line">            -M maskfile.gtf -u <span class="se">\</span>
</span><span class="line">            merged.gtf h1.bam
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash$">-p/--num-threads </code>：设置线程数，用于多核计算。</p>
  </li>
  <li>
    <p><code class="language-bash$">-o/--output-dir</code>：输出文件夹。</p>
  </li>
  <li>
    <p><code class="language-bash$">-b/--frag-bias-correct </code>：参考基因组序列，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-M/--mask-file </code>：过滤序列GTF文件，放入这个文件中的序列都不会被计算，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-u/--multi-read-correct</code>：处理map到基因组多个位置的reads，<strong>建议添加</strong>。</p>
  </li>
</ul>

<h3 id="cuffdifflanguage-bash-">2.4 <code class="language-bash">cuffdiff</code> 快速运行</h3>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code for cuffdiff </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># cuffdiff可以输入cuffquant生成的cxb文件或者原始的bam文件</span>
</span><span class="line"><span class="nv">$ </span>cuffdiff -p <span class="m">14</span> -o diffOut -b hg19.fa <span class="se">\</span>
</span><span class="line">           -M maskfile.gtf -u <span class="se">\</span>
</span><span class="line">           -L P,R merged.gtf h1.cxb,h2.cxb,h5.cxb h2.cxb,h4.cxb,h6.cxb
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash$">-p/--num-threads </code>：设置线程数，用于多核计算。</p>
  </li>
  <li>
    <p><code class="language-bash$">-o/--output-dir</code>：输出文件夹。</p>
  </li>
  <li>
    <p><code class="language-bash$">-b/--frag-bias-correct </code>：参考基因组序列，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-M/--mask-file </code>：过滤序列GTF文件，放入这个文件中的序列都不会被计算，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-u/--multi-read-correct</code>：处理map到基因组多个位置的reads，<strong>建议添加</strong>。</p>
  </li>
  <li>
    <p><code class="language-bash$">-L/--labels </code>：指定样品组名</p>
  </li>
</ul>

<h2 id="cummerbund">3. CummeRbund</h2>

<p>个人感觉CummeRbund不好用，不做介绍和讨论。</p>

<h3 id="section">参考资料</h3>

<ul>
  <li>
    <p><a href="http://www.chenlianfu.com/?p=623">一篇介绍cufflinks的中文博客</a></p>
  </li>
  <li>
    <p><a href="http://cole-trapnell-lab.github.io/cufflinks/getting_started/#common-uses-of-the-cufflinks-package">cufflinks安装介绍</a></p>
  </li>
  <li>
    <p><a id="Roberts et.al., 2011">Roberts A</a>, Pimentel H, Trapnell C, Pachter L: <strong>Identification of novel transcripts in annotated genomes using RNA-Seq.</strong> <em>Bioinformatics</em>. 2011, 27(17):2325-9. <a href="http://bioinformatics.oxfordjournals.org/content/early/2011/06/21/bioinformatics.btr355.full.pdf">pdf</a></p>
  </li>
  <li>
    <p><a href="http://www.ebiotrade.com/newsf/2015-3/2015316173503298.htm">RNA-seq分析新三件套</a></p>
  </li>
</ul>

<h3 id="section-1">更新记录</h3>

<p>2015年6月10日</p>
</div>


      <footer>
        <p class="meta text-muted">
          
  

<span class="glyphicon glyphicon-user"></span> <span class="byline author vcard" itemprop="author" itemscope itemtype="http://schema.org/Person">Posted by <span class="fn" itemprop="name">Yulong Niu</span></span>

          












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-06-26T14:51:56+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
          

<span class="glyphicon glyphicon-tags"></span>&nbsp;
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinfor/'>bioinfor</a>
  
</span>


        </p>
        
          <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund/" data-via="" data-counturl="http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund/" >Tweet</a>
  
  
  
</div>

        
        
          <ul class="meta text-muted pager">
            
            <li class="previous"><a href="/blog/2014/06/25/parallel-package/" title="Previous Post: R使用parallel包并行计算">&laquo; R使用parallel包并行计算</a></li>
            
            
            <li class="next"><a href="/blog/2014/07/05/my-first-post/" title="Next Post: My first post">My first post &raquo;</a></li>
            
          </ul>
        
      </footer>
    </article>
    
      <section>
        <h2>Comments</h2>
        <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
      </section>
    
  </div>

  
  <aside class="sidebar col-md-3">
    
      <section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Recent Posts</h3>
  </div>
  
  <div id="recent_posts" class="list-group">
    
    <a class="list-group-item " href="/blog/2018/07/19/rcppparallel-test/">使用RcppParallel并行计算</a>
    
    <a class="list-group-item " href="/blog/2018/02/17/estimate-rna-seq/">通过RNA-Seq评估基因表达量的模型</a>
    
    <a class="list-group-item " href="/blog/2017/10/24/bray-curtis-distance/">Bray-Curtis Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/22/google-distance/">Normalized Google Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/16/max-entropy/">最大熵模型</a>
    
  </div>
</section>
<section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Categories</h3>
  </div>
  <div class="list-group">
    
    
    <a class="list-group-item " href="/blog/categories/linux/index.html">
        <span class="badge">10</span>
        linux
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/r/index.html">
        <span class="badge">10</span>
        r
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/lisp/index.html">
        <span class="badge">6</span>
        lisp
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/ppr/index.html">
        <span class="badge">6</span>
        ppr
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/bioinfor/index.html">
        <span class="badge">22</span>
        bioinfor
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/saysomething/index.html">
        <span class="badge">1</span>
        saysomething
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/c/index.html">
        <span class="badge">2</span>
        c
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/hcj/index.html">
        <span class="badge">1</span>
        hcj
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/share/index.html">
        <span class="badge">1</span>
        share
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/java/index.html">
        <span class="badge">1</span>
        java
      </a>
    
  </div>
</section>






    
  </aside>
  
</div>

        </div>
      </div>
    </div>
    <footer role="contentinfo"><div class="container">
    <p class="text-muted credits">
  Copyright &copy; 2018 - Yulong Niu<br>
  <small>
      <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>,
      <span class="credit">customized with <a href="https://github.com/kAworu/octostrap3">octostrap3</a></span>.
  </small>
</p>

</div>
</footer>
    

<script type="text/javascript">
      var disqus_shortname = 'yulongniu';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund/';
        var disqus_url = 'http://yulongniu.bionutshell.org/blog/2014/06/26/tophat-cufflinks-cummerbund/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>


<script src="/assets/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="/javascripts/modernizr-2.0.js"></script>


  </body>
</html>
