<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
  <meta charset="utf-8">
  <title>清洗二代测序数据 - 牛牛龙</title>
  <meta name="author" content="Yulong Niu">

  
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data">
  <link href="/favicon.ico" type="image/png" rel="icon">
  <link href="/atom.xml" rel="alternate" title="牛牛龙" type="application/atom+xml">

  <!-- http://opengraphprotocol.org/ -->
  <meta name="twitter:card" content="summary_large_image">
  <meta property="og:type" content="website">
  <meta property="og:url" content="http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data">
  <meta property="og:title" content="清洗二代测序数据 - 牛牛龙">
  

  <!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({ TeX: { extensions: ["color.js"] }});
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

<script src="/javascripts/libs/jquery/jquery-2.0.3.min.js"></script>

<!--enable to choose theme in octostrap-->

  <link href="/assets/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/assets/bootstrap/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">



<!-- link open with new tab  -->
<script>
  function addBlankTargetForLinks () {
  $('a[href^="http"]').each(function(){
  $(this).attr('target', '_blank');
  });
  }
  
  $(document).bind('DOMNodeInserted', function(event) {
  addBlankTargetForLinks();
  });
</script>


  
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  

  
   <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-53073828-1', 'auto');
    ga('send', 'pageview');

  </script>


</head>

  <body   >
    <a href="#content" class="sr-only sr-only-focusable">Skip to main content</a>
    <div id="wrap">
      <header role="banner">
        <nav class="navbar navbar-default" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">牛牛龙</a>
        </div>

        <div class="navbar-collapse collapse">
            <ul class="nav navbar-nav">
                <li class="active">
                    <a rel="index" href="/">Blog</a>
                </li>
                <li >
                    <a href="/blog/archives">Archives</a>
                </li>
		<li >
                  <a href="/whoami">WhoAmI</a>
                </li>
		<li >
                  <a href="/link">Link</a>
                </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="subscribe-rss" href="/atom.xml" title="subscribe via RSS">
                        <span class="visible-xs">RSS</span>
                        <img class="hidden-xs" src="/images/rss.png" alt="RSS">
                    </a>
                </li>
                
            </ul>
            
                <form class="search navbar-form navbar-right" action="https://www.google.com/search" method="GET">
                    <input type="hidden" name="q" value="site:yulongniu.bionutshell.org">
                    <div class="form-group">
                        <input class="form-control" type="text" name="q" placeholder="Search">
                    </div>
                </form>
            
        </div>
    </div>
</nav>


      </header>
      <div id="main" role="main" class="container">
        <div id="content">
          <div class="row">
  <div class="page-content col-md-9" itemscope itemtype="http://schema.org/Blog">
    <meta itemprop="name" content="牛牛龙" />
    
    <meta itemprop="url" content="http://yulongniu.bionutshell.org" />
    <article class="hentry" role="article" itemprop="blogPost" itemscope itemtype="http://schema.org/BlogPosting">
      
  <header class="page-header">
    
      <p class="meta text-muted text-uppercase">
        












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-28T20:11:18+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
        
           | <a href="#disqus_thread" itemprop="discussionUrl"
             data-disqus-identifier="http://yulongniu.bionutshell.org">Comments</a>
        
      </p>
    
    
    <h1 class="entry-title" itemprop="name headline">
        清洗二代测序数据
        
    </h1>
    
  </header>


<div class="entry-content clearfix" itemprop="articleBody description"><h2 id="fastqc">1. FastQC</h2>

<p><strong>简介</strong>：<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>是用于对二代测序数据质量快速检验的工具，可以输入fastq（fastq.gz）、sam或者bam文件。查看<a href="http://www.bioinformatics.nl/courses/RNAseq/FastQC_Manual.pdf">输出结果解释</a>。</p>

<!--more-->

<p><strong>平台</strong>：所有平台。</p>

<p><strong>安装</strong>： 依赖Java，下载后直接安装使用。</p>

<p><strong>快速运行</strong>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 输出分析结果至特定文档</span>
</span><span class="line"><span class="nv">$ </span>fastqc seqFile1 --outdir setFolder1
</span><span class="line">
</span><span class="line"><span class="c"># 支持批量处理测序数据</span>
</span><span class="line"><span class="nv">$ </span>fastqc seqFile1 seqFile2 seqFileN
</span><span class="line">
</span><span class="line"><span class="c"># 查看帮助信息</span>
</span><span class="line"><span class="nv">$ </span>fastqc --help
</span><span class="line">
</span><span class="line"><span class="c"># 查看一共分析了多少个reads，比如fastqc文件为“accepted_filtered_fastqc.zip”</span>
</span><span class="line"><span class="nv">$ </span>unzip -p accepted_filtered_fastqc.zip accepted_filtered_fastqc/fastqc_data.txt <span class="p">|</span> <span class="se">\</span>
</span><span class="line">      sed -n <span class="s1">&#39;7 p&#39;</span> <span class="p">|</span> <span class="se">\</span>
</span><span class="line">      awk <span class="s1">&#39;{print $3}&#39;</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h2 id="trim-galore">2. Trim Galore!</h2>

<p><strong>简介</strong>：<a href="http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore!</a>是对<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>和<a href="https://cutadapt.readthedocs.org/en/stable/installation.html">Cutadapt</a>的包装。可以处理Illumina、Nextera 3和smallRNA测序平台的双端和单端数据，包括去除adapter和低质量reads。</p>

<p><strong>平台</strong>：Linux</p>

<p><strong>安装</strong>:</p>

<ul>
  <li>需要先分别安装FastQC和Cutadapt，其中Cutadapt安装使用</li>
</ul>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Install Cutadapt</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># pip install cutadapt</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>快速运行</strong>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code of runing the Cutadapt</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 处理双端测序结果</span>
</span><span class="line"><span class="nv">$ </span>trim_galore --quality <span class="m">20</span> --phred33 --stringency <span class="m">3</span> --length <span class="m">20</span> --paired <span class="se">\</span>
</span><span class="line">              --gzip --output_dir human_trimgalore <span class="se">\</span>
</span><span class="line">              mySeq_1_1.fastq.gz mySeq_1_2.fastq.gz
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash$">--quality</code>：设定Phred quality score阈值，默认为20。</p>
  </li>
  <li>
    <p><code class="language-bash$">--phred33</code>：：选择<code class="language-bash">-phred33</code>或者<code class="language-bash">-phred64</code>，表示测序平台使用的Phred quality score。</p>
  </li>
  <li>
    <p><code class="language-bash$">--adapter</code>：输入adapter序列。也可以不输入，Trim Galore!会自动寻找可能性最高的平台对应的adapter。自动搜选的平台三个，也直接显式输入这三种平台，即<code class="language-bash">--illumina</code>、<code class="language-bash">--nextera</code>和<code class="language-bash">--small_rna</code>。</p>
  </li>
  <li>
    <p><code class="language-bash$">--stringency</code>：设定可以忍受的前后adapter重叠的碱基数，默认为1（非常苛刻）。可以适度放宽，因为后一个adapter几乎不可能被测序仪读到。</p>
  </li>
  <li>
    <p><code class="language-bash$">--length</code>：设定输出reads长度阈值，小于设定值会被抛弃。</p>
  </li>
  <li>
    <p><code class="language-bash$">--paired</code>：对于双端测序结果，一对reads中，如果有一个被剔除，那么另一个会被同样抛弃，而不管是否达到标准。</p>
  </li>
  <li>
    <p><code class="language-bash$">--retain_unpaired</code>：对于双端测序结果，一对reads中，如果一个read达到标准，但是对应的另一个要被抛弃，达到标准的read会被单独保存为一个文件。</p>
  </li>
  <li>
    <p><code class="language-bash$">--gzip</code>和<code class="language-bash$">--dont_gzip</code>：清洗后的数据zip打包或者不打包。</p>
  </li>
  <li>
    <p><code class="language-bash$">--output_dir</code>：输入目录。需要提前建立目录，否则运行会报错。</p>
  </li>
</ul>

<h2 id="trimmomatic">3. Trimmomatic</h2>

<p><strong>简介</strong>：<a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>是针对Illumina高通量测序平台设计的接头去除和低质量reads清洗软件。软件中包括有Illumina平台常见接头序列，可以很方便处理单端和双端RNA-seq数据。Trimmomatic也支持自己设计要去除的接头序列文件。</p>

<p><strong>平台</strong>：Java跨平台使用</p>

<p><strong>快速运行</strong>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Example code of runing the Trimmomatic </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 处理双端测序结果</span>
</span><span class="line"><span class="nv">$ </span>java -jar /path/trimmomatic-0.33.jar PE<span class="se">\</span>
</span><span class="line">       -threads <span class="m">12</span> -phred33 -trimlog mySeq_1-trim.log <span class="se">\</span>
</span><span class="line">       mySeq_1_1.fastq.gz mySeq_1_2.fastq.gz <span class="se">\</span>
</span><span class="line">       mySeq_1_1-trim.fastq.gz mySeq_1_1-unpair.fastq.gz <span class="se">\</span>
</span><span class="line">       mySeq_1_2-trim.fastq.gz mySeq_1_2-unpair.fastq.gz <span class="se">\</span>
</span><span class="line">       ILLUMINACLIP:/path/TruSeq3-PE.fa:2:30:10 <span class="se">\</span>
</span><span class="line">       LEADING:3 <span class="se">\</span>
</span><span class="line">       TRAILING:3 <span class="se">\</span>
</span><span class="line">       SLIDINGWINDOW:4:15 <span class="se">\</span>
</span><span class="line">       MINLEN:51
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li>
    <p><code class="language-bash">-threads</code>：设置线程数目。</p>
  </li>
  <li>
    <p><code class="language-bash">-phred33</code>：选择<code class="language-bash">-phred33</code>或者<code class="language-bash">-phred64</code>，表示测序平台使用的Phred quality score。查询方法：首先，运行<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>，在结果报告第一项会“猜出”测序平台。之后，查询平台对应Phred<a href="http://en.wikipedia.org/wiki/FASTQ_format">列表</a>。</p>
  </li>
  <li>
    <p><code class="language-bash">-trimlog</code>：输出运行日志。日志中包括对每一个read具体选择数据，所以文件会比较大。</p>
  </li>
  <li>
    <p><code class="language-bash">ILLUMINACLIP</code>：跟随四个参数，分别是<code class="language-bash">:&lt;fastaWithAdaptersEtc&gt;</code>为adaptesr文件完整路径（在Trimmomatic的默认安装目录下的 <code class="language-bash">adapter</code>，有整理好的）；<code class="language-bash">&lt;fastaWithAdaptersEtc&gt;</code>为seed matches（16bases）在匹配时的最大错配数目；<code class="language-bash">&lt;palindrome clip threshold&gt;</code>对于一对reads当得分超过30（约50 bases），seeds会被延伸和固定；<code class="language-bash">&lt;simple clip threshold&gt;</code>，对于单端reads当得分超过10（约17 bases），seeds会被延伸和固定。</p>
  </li>
  <li>
    <p><code class="language-bash">LEADING</code>和<code class="language-bash">TRAILING</code>：分别为去除read头部和尾部的低质量（低于quality3）碱基数目。</p>
  </li>
  <li>
    <p><code class="language-bash">SLIDINGWINDOW</code>：跟随两个参数，分别是 <code class="language-bash">&lt;windowSize&gt;</code>为扫描“窗口”长度；<code class="language-bash">&lt;requiredQuality&gt;</code>为窗口碱基质量的平均阈值，低于此会被删除。</p>
  </li>
  <li>
    <p><code class="language-bash">MINLEN</code>：设置最短reads数目。需要根据下游alignment软件设定，比如<a href="http://bowtie-bio.sourceforge.net/index.shtml">Bowtie</a>适用于短序列，比如50bp以下；而<a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a>适用于50bp以上。<a href="http://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a> 则根据实际使用Bowtie或者Bowtie2选择。</p>
  </li>
</ul>

<h3 id="section">更新记录</h3>

<p>2016年9月10日</p>
</div>


      <footer>
        <p class="meta text-muted">
          
  

<span class="glyphicon glyphicon-user"></span> <span class="byline author vcard" itemprop="author" itemscope itemtype="http://schema.org/Person">Posted by <span class="fn" itemprop="name">Yulong Niu</span></span>

          












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-28T20:11:18+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
          

<span class="glyphicon glyphicon-tags"></span>&nbsp;
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinfor/'>bioinfor</a>
  
</span>


        </p>
        
          <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data/" data-via="" data-counturl="http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data/" >Tweet</a>
  
  
  
</div>

        
        
          <ul class="meta text-muted pager">
            
            <li class="previous"><a href="/blog/2014/07/26/short-sequence-alignment/" title="Previous Post: 二代测序中的短序列比对">&laquo; 二代测序中的短序列比对</a></li>
            
            
            <li class="next"><a href="/blog/2014/12/02/emacs-config-c/" title="Next Post: Emacs配置C语言编程环境">Emacs配置C语言编程环境 &raquo;</a></li>
            
          </ul>
        
      </footer>
    </article>
    
      <section>
        <h2>Comments</h2>
        <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
      </section>
    
  </div>

  
  <aside class="sidebar col-md-3">
    
      <section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Recent Posts</h3>
  </div>
  
  <div id="recent_posts" class="list-group">
    
    <a class="list-group-item " href="/blog/2017/06/14/shadowsocks-setup/">Shadowsocks设置推荐</a>
    
    <a class="list-group-item " href="/blog/2017/05/16/java-binary/">探索Java基本类型的二进制表示</a>
    
    <a class="list-group-item " href="/blog/2016/11/11/git-rebase-examples/">两个git的rebase命令应用</a>
    
    <a class="list-group-item " href="/blog/2016/10/21/linux-sync-google-drive/">命令行使用Google Drive</a>
    
    <a class="list-group-item " href="/blog/2016/09/22/how-to-publish-in-nc/">如何在Nature Communications上发表文章</a>
    
  </div>
</section>
<section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Categories</h3>
  </div>
  <div class="list-group">
    
    
    <a class="list-group-item " href="/blog/categories/linux/index.html">
        <span class="badge">9</span>
        linux
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/r/index.html">
        <span class="badge">9</span>
        r
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/lisp/index.html">
        <span class="badge">6</span>
        lisp
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/ppr/index.html">
        <span class="badge">5</span>
        ppr
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/bioinfor/index.html">
        <span class="badge">13</span>
        bioinfor
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/saysomething/index.html">
        <span class="badge">1</span>
        saysomething
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/c/index.html">
        <span class="badge">2</span>
        c
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/hcj/index.html">
        <span class="badge">1</span>
        hcj
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/share/index.html">
        <span class="badge">1</span>
        share
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/java/index.html">
        <span class="badge">1</span>
        java
      </a>
    
  </div>
</section>






    
  </aside>
  
</div>

        </div>
      </div>
    </div>
    <footer role="contentinfo"><div class="container">
    <p class="text-muted credits">
  Copyright &copy; 2017 - Yulong Niu<br>
  <small>
      <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>,
      <span class="credit">customized with <a href="https://github.com/kAworu/octostrap3">octostrap3</a></span>.
  </small>
</p>

</div>
</footer>
    

<script type="text/javascript">
      var disqus_shortname = 'yulongniu';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data/';
        var disqus_url = 'http://yulongniu.bionutshell.org/blog/2014/07/28/clean-seq-data/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>


<script src="/assets/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="/javascripts/modernizr-2.0.js"></script>


  </body>
</html>
