<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
  <meta charset="utf-8">
  <title>二代测序中的常用工具介绍 - 牛牛龙</title>
  <meta name="author" content="Yulong Niu">

  
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools">
  <link href="/favicon.ico" type="image/png" rel="icon">
  <link href="/atom.xml" rel="alternate" title="牛牛龙" type="application/atom+xml">

  <!-- http://opengraphprotocol.org/ -->
  <meta name="twitter:card" content="summary_large_image">
  <meta property="og:type" content="website">
  <meta property="og:url" content="http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools">
  <meta property="og:title" content="二代测序中的常用工具介绍 - 牛牛龙">
  

  <!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
     tex2jax: {
         inlineMath: [ ['$','$'], ["\\(","\\)"] ],
         processEscapes: true
     }
 });
</script>

<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
     tex2jax: {
         skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
     }
 });
</script>

<script type="text/x-mathjax-config">
 MathJax.Hub.Queue(function() {
     var all = MathJax.Hub.getAllJax(), i;
     for(i=0; i < all.length; i += 1) {
         all[i].SourceElement().parentNode.className += ' has-jax';
     }
 });
</script>

<script type="text/javascript"
        src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

<script type="text/x-mathjax-config">
 MathJax.Hub.Queue(function() {
     var all = MathJax.Hub.getAllJax(), i;
     for(i=0; i < all.length; i += 1) {
         all[i].SourceElement().parentNode.className += ' has-jax';
     }
 });
</script>

<script src="/javascripts/libs/jquery/jquery-2.0.3.min.js"></script>

<!--enable to choose theme in octostrap-->

  <link href="/assets/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/assets/bootstrap/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">



<!-- link open with new tab  -->
<script>
  function addBlankTargetForLinks () {
  $('a[href^="http"]').each(function(){
  $(this).attr('target', '_blank');
  });
  }

  $(document).bind('DOMNodeInserted', function(event) {
  addBlankTargetForLinks();
  });
</script>


  
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  

  
   <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-53073828-1', 'auto');
    ga('send', 'pageview');

  </script>


</head>

  <body   >
    <a href="#content" class="sr-only sr-only-focusable">Skip to main content</a>
    <div id="wrap">
      <header role="banner">
        <nav class="navbar navbar-default" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">牛牛龙</a>
        </div>

        <div class="navbar-collapse collapse">
            <ul class="nav navbar-nav">
                <li class="active">
                    <a rel="index" href="/">Blog</a>
                </li>
                <li >
                    <a href="/blog/archives">Archives</a>
                </li>
		<li >
                  <a href="/whoami">WhoAmI</a>
                </li>
		<li >
                  <a href="/link">Link</a>
                </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="subscribe-rss" href="/atom.xml" title="subscribe via RSS">
                        <span class="visible-xs">RSS</span>
                        <img class="hidden-xs" src="/images/rss.png" alt="RSS">
                    </a>
                </li>
                
            </ul>
            
                <form class="search navbar-form navbar-right" action="https://www.google.com/search" method="GET">
                    <input type="hidden" name="q" value="site:yulongniu.bionutshell.org">
                    <div class="form-group">
                        <input class="form-control" type="text" name="q" placeholder="Search">
                    </div>
                </form>
            
        </div>
    </div>
</nav>


      </header>
      <div id="main" role="main" class="container">
        <div id="content">
          <div class="row">
  <div class="page-content col-md-9" itemscope itemtype="http://schema.org/Blog">
    <meta itemprop="name" content="牛牛龙" />
    
    <meta itemprop="url" content="http://yulongniu.bionutshell.org" />
    <article class="hentry" role="article" itemprop="blogPost" itemscope itemtype="http://schema.org/BlogPosting">
      
  <header class="page-header">
    
      <p class="meta text-muted text-uppercase">
        












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-29T00:23:42+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
        
           | <a href="#disqus_thread" itemprop="discussionUrl"
             data-disqus-identifier="http://yulongniu.bionutshell.org">Comments</a>
        
      </p>
    
    
    <h1 class="entry-title" itemprop="name headline">
        二代测序中的常用工具介绍
        
    </h1>
    
  </header>


<div class="entry-content clearfix" itemprop="articleBody description"><h2 id="samtools">1. SAMtools</h2>

<p><strong>简介</strong>：<a href="http://www.htslib.org/">SAMtools</a>是用于处理SAM (Sequence Alignment/Map)格式文件的一系列工具，主要用来储存大容量的核酸测序结果。BAM是SAM文件的binary格式文件。SAMtools的主要作者是<a href="http://lh3lh3.users.sourceforge.net/">Heng Li</a>，Heng Li在2012年因为对二代测序领域的贡献获得<a href="http://en.wikipedia.org/wiki/Benjamin_Franklin_Award_(Bioinformatics)">Benjamin Franklin Award</a>。</p>

<p><strong>平台</strong>：Mac OS/Linux</p>

<p><strong>安装</strong>:</p>

<ul>
  <li><a href="http://www.htslib.org/download/">下载</a>SAMtools</li>
</ul>

<!--more-->

<ul>
  <li>安装依赖的Ncurse库</li>
</ul>

<div class="bogus-wrapper"><notextile><figure class="code"><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># yum install ncurses*</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<ul>
  <li>添加SAMtools到PATH（<a href="http://yulongniu.bionutshell.org/blog/2010/11/08/linux-command/">设置参考</a>）</li>
</ul>

<p><strong>快速运行</strong></p>

<div class="bogus-wrapper"><notextile><figure class="code"><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># sam格式文件转换为bam格式文件</span>
</span><span class="line"><span class="c"># 新版本samtools不用使用-S</span>
</span><span class="line"><span class="nv">$ </span>samtools view -b samFile &gt; bamFile
</span><span class="line">
</span><span class="line"><span class="c"># bam格式文件转换为sam格式文件</span>
</span><span class="line"><span class="nv">$ </span>samtools view -h bamFile &gt; samFile
</span><span class="line">
</span><span class="line"><span class="c"># bam格式文件转换为sorted bam格式文件，用于长期储存和后续分析</span>
</span><span class="line"><span class="c"># 后一个‘sortedBamFilePrefix’是指需要存储文件名前缀，比如想存储“human_1.bam”，则输入“human_1”</span>
</span><span class="line"><span class="nv">$ </span>samtools sort bamFile sortedBamFilePrefix
</span><span class="line">
</span><span class="line"><span class="c"># bam文件按照reads名称排序</span>
</span><span class="line"><span class="nv">$ </span>samtools sort -n bamFile sortedBamFilePrefix
</span><span class="line">
</span><span class="line"><span class="c"># 直接查看bam文件</span>
</span><span class="line"><span class="nv">$ </span>samtools view bamFile <span class="p">|</span> head -2
</span><span class="line">
</span><span class="line"><span class="c"># 创建bam的index文件</span>
</span><span class="line"><span class="nv">$ </span>samtools index bamFile
</span><span class="line">
</span><span class="line"><span class="c"># 输出alignment数目，配合-f和-F过滤reads</span>
</span><span class="line"><span class="nv">$ </span>samtools view -c bamFile
</span><span class="line">
</span><span class="line"><span class="c"># 统计bam文件map数量，可以用于评估mapping的质量。需要输入indexed的bam文件</span>
</span><span class="line"><span class="c"># 输出结果每一列分别为参考序列名称、参考序列长度、map上的reads数目、未map上的reads数目</span>
</span><span class="line"><span class="nv">$ </span>samtools idxstats sortedIndexedBamFile
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>重要参数解释</strong>：</p>

<ul>
  <li><code class="language-bash">view -h</code>：打印bam文件头部，文件头部信息用于转换sam文件至bam文件。<code class="language-bash">view -H</code>：只打印bam文件头部。</li>
</ul>

<p><strong>补充</strong>：</p>

<ul>
  <li>一个sam文件式例子</li>
</ul>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>A small sample of sam file</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="nv">$ </span>head samFile
</span><span class="line">HD      VN:1.0          SO:coordinate
</span><span class="line">@SQ     SN:chr1         LN:249250621
</span><span class="line">@SQ     SN:chr10        LN:135534747
</span><span class="line">@SQ     SN:chr11        LN:135006516
</span><span class="line">
</span><span class="line">HISEQ2000-02:436:C2PG3ACXX:3:1316:4503:75198    <span class="m">99</span>      chr1    <span class="m">11527</span>   <span class="m">3</span>       <span class="nv">100M</span>    <span class="o">=</span>       <span class="m">11624</span>   <span class="m">197</span>     CTGGGTTTAAAAGTAAAAAATAAATATGTTTAATTTGTGAACTGATTACCATCAGAATTGTACTGTTCTGTATCCCACCAGCAATGTCTAGGAATGCCTG  ?@@FFFD?D&gt;DDHFGGEGGIJDHJCEGCHE@FCHIJJJGG@GEGHGJGHHIIIJJJIJJECGG@FHIIJJJIIGEADHHEEDF<span class="p">;</span>BB@CACCCEDDDDDAC    AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:100     YT:Z:UU  NH:i:2  CC:Z:chr15      CP:i:102519544  HI:i:0
</span><span class="line">
</span><span class="line">HISEQ2000-06:325:C2RC0ACXX:5:1101:16521:147635  <span class="m">355</span>     chr1    <span class="m">11554</span>   <span class="m">0</span>       <span class="nv">100M</span>    <span class="o">=</span>       <span class="m">11681</span>   <span class="m">227</span>     GTTTAATTTGTTAACTGATTACCATCAGAATTGTACTGTTCTGTATCCCACCAGCAATGTCTAGGAATGCCTGTTTCTCCACAAAGTGTTTACTTTTGGA  @@@DBDDDHFDDDHGGBA@AEFBBCGIC?BCGIAHBEFFHIHIDHC@DEGIIIHEG<span class="p">;</span>8BDBFGGICFI:FG7@FHEHIICIEA<span class="p">;</span>CA<span class="p">;</span>77<span class="p">;</span>7<span class="p">;;</span>@<span class="p">;</span>@A@6<span class="o">(</span>    AS:i:-5 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:11G88   YT:Z:UU  NH:i:5  CC:Z:chr12      CP:i:93956      HI:i:0
</span><span class="line">
</span><span class="line">HISEQ2000-06:325:C2RC0ACXX:5:2304:4393:52082    <span class="m">163</span>     chr1    <span class="m">11579</span>   <span class="m">3</span>       <span class="nv">100M</span>    <span class="o">=</span>       <span class="m">11643</span>   <span class="m">164</span>     CAGAATTGTACTGTTCTGTATCCCACCAGCAATGTCTAGGAATACCTGTTTCTCCACAAAGTGTTTACTTTTGGATTTTTGCCAGTCTAACAGGTGAAGC  CCCFFFFFHHHHHIJJJJJJJJJJJJJJJJJJJJIJJJJJJJJJJJJJJJGIJJJIJJJJJFHFHHHIJJJJJIEHIJHHHHFFFFFFEEEEEDDDDDDD    AS:i:-6 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:43G56   YT:Z:UU  NH:i:2  CC:Z:chr15      CP:i:102519492  HI:i:0
</span></code></pre></td></tr></table></div></figure></notextile></div>

<ul>
  <li>
    <p>sam文件内容各行解释，参考<a href="http://samtools.github.io/hts-specs/">SAM/BAM and related specifications</a>和<a href="http://blog.qiuworld.com:8080/archives/3050">NGS分析入门：操作SAM/BAM文件</a>。</p>
  </li>
  <li>
    <p>bam文件过滤，参考<a href="http://blog.qiuworld.com:8080/archives/3419">如何统计BAM文件中的reads数</a></p>
  </li>
</ul>

<h2 id="bowtie2">2. Bowtie2</h2>

<p>Bowtie使用介绍，详见<a href="http://yulongniu.bionutshell.org/blog/2014/07/26/short-sequence-alignment/">二代测序中的短序列比对</a> 。</p>

<h2 id="trinity">3. Trinity</h2>

<p><a href="http://trinityrnaseq.github.io/">Trinity</a>是<a href="http://www.broadinstitute.org/">Broad Institute</a>开发的根据RNA-seq数据从头（<em>de novo</em>）组装转录组的工具。</p>

<h2 id="mirdeep2">4. miRDeep2</h2>

<p><a href="https://www.mdc-berlin.de/36105849/en/research/research_teams/systems_biology_of_gene_regulatory_elements/projects/miRDeep/documentation#mapper">miRDeep2</a> 是用于高通量测序中检测<a href="http://en.wikipedia.org/wiki/MicroRNA">miRNA</a> 的工具。</p>

<h2 id="miso">5. MISO</h2>

<p><strong>简介</strong>：<a href="http://genes.mit.edu/burgelab/miso/">MISO</a>（Mixture-of-Isoforms）是用来计算和探测RNA-seq数据中不同样本的基因选择剪切。</p>

<p><strong>平台</strong>：Python跨平台使用。</p>

<p><strong>快速运行</strong>：</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>An Example of Running MISO</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line">
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h2 id="section">6. 质量检测</h2>

<h3 id="rna-seqc">6.1 RNA-SeQC</h3>

<p><a href="http://www.broadinstitute.org/cancer/cga/rna-seqc">RNA-SeQC</a>是用于检测RNA-seq测序质量。</p>

<h3 id="rseqc">6.2 RSeQC</h3>

<p><a href="http://rseqc.sourceforge.net/">RSeQC</a>用于对RNA-seq数据质量控制。</p>

<h2 id="section-1">7. 小工具集锦</h2>

<h3 id="fastq">7.1 操作fastq文件</h3>

<p>在Linux机器上配以一系列命令，操作fastq或者fastq.gz文件，比如<code>mySeq_1_1.fastq</code>测序文件。</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Manipulating fastq file with bash </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># Fastq文件前几行</span>
</span><span class="line"><span class="nv">$ </span>head mySeq_1_1.fastq
</span><span class="line">@HISEQ2000-06:325:C2RC0ACXX:5:1101:1217:2215 1:N:0:TGCCGGCT
</span><span class="line">GGCACAGTCCATGCTTTTAACCAGATTTGAACAGAAGAATGGCCACTTGNNNNNNGTAGNNNNNNANNNNGNNNTNNNTNNCATGTGTCACATAACTACC
</span><span class="line">+
</span><span class="line">@C@FFFFDFFHHHGFFIIBEGHJIFHIIGCGIJJIJIJJIJIJJIIIGI######-.&lt;B######-####,###,###<span class="o">(</span><span class="c">##,,5&lt;?CDDDDCCDDDCDDD</span>
</span><span class="line">@HISEQ2000-06:325:C2RC0ACXX:5:1101:1621:2040 1:N:0:TGCCGGCT
</span><span class="line">NGGTGACCTTCTCCCGCCAGAAGCCAAAAACTGCAGCCTACTTTTCTGAAGTGGTTATCTTGGGACTGAGGTATGGGCTATCTTGGGCTGTTTCCTATTT
</span><span class="line">+
</span><span class="line"><span class="c">#1==D;ADBHDFDHFHGGIIB;FFHIIGGGIGADEGEIIAHEEHIGHH==C8BGC;=FCDHICHCAAC&gt;CE..;A&gt;AC?9@CCCCCBBCB?CCCCCCCCC</span>
</span><span class="line">@HISEQ2000-06:325:C2RC0ACXX:5:1101:1663:2146 1:N:0:TGCCGGCT
</span><span class="line">AGAGCCAAATATTTCAACAAAACTGCAGTTTAATTTCAGAAAATGTTAAAATATATATTTATACATCAATTTCTGACATACACTTAATGTGTTAGTATAC
</span><span class="line">
</span><span class="line"><span class="c"># 统计fastq文件中的reads数量</span>
</span><span class="line"><span class="nv">$ </span>awk <span class="s1">&#39;END{print NR/4}&#39;</span> mySeq_1_1.fastq
</span><span class="line">
</span><span class="line"><span class="c"># 查看fastq.gz文件</span>
</span><span class="line"><span class="nv">$ </span>zcat mySeq_1_1.fastq.gz
</span><span class="line">
</span><span class="line"><span class="c"># 统计fastq.gz文件reads数量</span>
</span><span class="line"><span class="c"># 以下两种方法等价</span>
</span><span class="line"><span class="nv">$ </span>zcat mySeq_1_1.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;END{print NR/4}&#39;</span>
</span><span class="line"><span class="nv">$ </span>zcat mySeq_1_1.fastq.gz <span class="p">|</span> <span class="nb">echo</span> <span class="k">$((</span><span class="sb">`</span>wc -l<span class="sb">`</span><span class="o">/</span><span class="m">4</span><span class="k">))</span>
</span><span class="line"><span class="c"># 同时，由于一些测序文件会对read做特殊标记，比如“@HISEQ2000”。所以，可以使用正则匹配。</span>
</span><span class="line"><span class="c"># 注意：这种方法有潜在错误，慎重使用。</span>
</span><span class="line"><span class="nv">$ </span>zgrep -c <span class="s1">&#39;@HISEQ200&#39;</span> mySeq_1_1.fastq
</span><span class="line">
</span><span class="line"><span class="c"># 统计fastq文件reads长度分布</span>
</span><span class="line"><span class="nv">$ </span>zcat mySeq_1_1.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{if(NR%4==2) print length($1)}&#39;</span> <span class="p">|</span> sort -n <span class="p">|</span> uniq -c
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h3 id="fasta">7.2 操作fasta文件</h3>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Manipulating fasta file with bash </span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 序列数目</span>
</span><span class="line"><span class="nv">$ </span>grep -c <span class="s1">&#39;&gt;@&#39;</span> mySeq.fa
</span><span class="line">
</span><span class="line"><span class="c"># 提取特定序列</span>
</span><span class="line">samtools faidx mySeq.fa
</span><span class="line">samtools faidx mySeq.fa chr1 chr2 chr3
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h3 id="picard">7.3 Picard</h3>

<p><a href="http://broadinstitute.github.io/picard/">Picard</a>是一个Java平台的工具包，包括一系列处理高通量测序的命令行工具。</p>

<h3 id="fastx-toolkit">7.4 FASTX-Toolkit</h3>

<p><a href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-Toolkit</a>是一系列用于处理大量短序列FASTA和FASTQ文件的工具。</p>

<h2 id="section-2">8. 各种数据类型</h2>

<p>为了下游分析，高通量测序结果往往使用多种数据格式储存，详细参考<a href="http://www.broadinstitute.org/igv/?q=book/export/html/16">File Formats</a>。</p>

<ul>
  <li>GTF格式文件，比如<a href="http://mblab.wustl.edu/GTF22.html">Ensembl类型</a>的GTF文件，每一列<a href="http://www.gencodegenes.org/gencodeformat.html">说明</a>、生物学序列分类<a href="http://www.gencodegenes.org/gencode_biotypes.html">GENCODE解释</a>和<a href="http://vega.sanger.ac.uk/info/about/gene_and_transcript_types.html">VEGA解释</a>。GTF文件的第二行中，有<a href="http://en.wikipedia.org/wiki/Coding_region">CDS（Coding DNA Sequence）</a>也有<a href="http://en.wikipedia.org/wiki/Exon">exon</a>，这两者概念不同：CDS只包括翻译成蛋白质的序列；exon也包括<a href="http://en.wikipedia.org/wiki/Untranslated_region">UTR</a>区域和ployA区域；exon也可以用来指示非编码RNA。</li>
</ul>

<h3 id="gtfgff">8.1 GTF和GFF文件互转</h3>

<p>使用<a href="http://cole-trapnell-lab.github.io/cufflinks/file_formats/">Cufflinks</a>的<code class="language-bash">gffread</code>直接进行转换。</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Convert the File Format Between GTF and GFF</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># GTF转换为GFF/GFF3</span>
</span><span class="line"><span class="nv">$ </span>gffreads myGtfFile.gtf -o myGffFile.gff3
</span><span class="line">
</span><span class="line"><span class="c"># GFF/GFF3转换为GTF</span>
</span><span class="line"><span class="nv">$ </span>gffread myGffFile.gff3 -T -o myGtfFile.gtf
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h3 id="gtf">8.2 GTF文件提取序列信息</h3>

<p>使用<a href="http://cole-trapnell-lab.github.io/cufflinks/file_formats/">Cufflinks</a>的<code class="language-bash">gffread</code>直接进行转换。</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>Extract sequences from GTF files</span></figcaption>
<div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="bash"><span class="line"><span class="c"># 从GTF文件中提取序列信息</span>
</span><span class="line"><span class="nv">$ </span>gffread -w myFastaFile.fa -g myGenome.fa myGtfFile.gtf
</span></code></pre></td></tr></table></div></figure></notextile></div>

<h3 id="section-3">参考资料</h3>

<ul>
  <li>
    <p><a href="http://blog.qiuworld.com:8080/archives/3050">NGS分析入门：操作SAM/BAM文件</a></p>
  </li>
  <li>
    <p><a href="http://bioinformatics.cvr.ac.uk/blog/essential-awk-commands-for-next-generation-sequence-analysis/">Essential AWK Commands for Next Generation Sequence Analysis</a></p>
  </li>
</ul>

<h3 id="section-4">更新记录</h3>

<p>2018年4月3日</p>
</div>


      <footer>
        <p class="meta text-muted">
          
  

<span class="glyphicon glyphicon-user"></span> <span class="byline author vcard" itemprop="author" itemscope itemtype="http://schema.org/Person">Posted by <span class="fn" itemprop="name">Yulong Niu</span></span>

          












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-07-29T00:23:42+08:00"  data-updated="true" itemprop="datePublished dateCreated">ordinal</time>
          

<span class="glyphicon glyphicon-tags"></span>&nbsp;
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinfor/'>bioinfor</a>
  
</span>


        </p>
        
          <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools/" data-via="" data-counturl="http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools/" >Tweet</a>
  
  
  
</div>

        
        
          <ul class="meta text-muted pager">
            
            <li class="previous"><a href="/blog/2014/07/28/clean-seq-data/" title="Previous Post: 清洗二代测序数据">&laquo; 清洗二代测序数据</a></li>
            
            
            <li class="next"><a href="/blog/2014/12/02/emacs-config-c/" title="Next Post: Emacs配置C语言编程环境">Emacs配置C语言编程环境 &raquo;</a></li>
            
          </ul>
        
      </footer>
    </article>
    
      <section>
        <h2>Comments</h2>
        <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
      </section>
    
  </div>

  
  <aside class="sidebar col-md-3">
    
      <section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Recent Posts</h3>
  </div>
  
  <div id="recent_posts" class="list-group">
    
    <a class="list-group-item " href="/blog/2018/07/19/rcppparallel-test/">使用RcppParallel并行计算</a>
    
    <a class="list-group-item " href="/blog/2018/02/17/estimate-rna-seq/">通过RNA-Seq评估基因表达量的模型</a>
    
    <a class="list-group-item " href="/blog/2017/10/24/bray-curtis-distance/">Bray-Curtis Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/22/google-distance/">Normalized Google Distance解释</a>
    
    <a class="list-group-item " href="/blog/2017/10/16/max-entropy/">最大熵模型</a>
    
  </div>
</section>
<section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Categories</h3>
  </div>
  <div class="list-group">
    
    
    <a class="list-group-item " href="/blog/categories/linux/index.html">
        <span class="badge">10</span>
        linux
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/r/index.html">
        <span class="badge">10</span>
        r
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/lisp/index.html">
        <span class="badge">6</span>
        lisp
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/ppr/index.html">
        <span class="badge">6</span>
        ppr
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/bioinfor/index.html">
        <span class="badge">22</span>
        bioinfor
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/saysomething/index.html">
        <span class="badge">1</span>
        saysomething
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/c/index.html">
        <span class="badge">2</span>
        c
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/hcj/index.html">
        <span class="badge">1</span>
        hcj
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/share/index.html">
        <span class="badge">1</span>
        share
      </a>
    
    
    <a class="list-group-item " href="/blog/categories/java/index.html">
        <span class="badge">1</span>
        java
      </a>
    
  </div>
</section>






    
  </aside>
  
</div>

        </div>
      </div>
    </div>
    <footer role="contentinfo"><div class="container">
    <p class="text-muted credits">
  Copyright &copy; 2018 - Yulong Niu<br>
  <small>
      <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>,
      <span class="credit">customized with <a href="https://github.com/kAworu/octostrap3">octostrap3</a></span>.
  </small>
</p>

</div>
</footer>
    

<script type="text/javascript">
      var disqus_shortname = 'yulongniu';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools/';
        var disqus_url = 'http://yulongniu.bionutshell.org/blog/2014/07/29/next-seqence-tools/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>


<script src="/assets/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="/javascripts/modernizr-2.0.js"></script>


  </body>
</html>
